app/src/main/java/space/taran/arknavigator/utils/FileUtils.kt

```diff
@@ -18,38 +20,48 @@ enum class Sorting {
     DEFAULT, NAME, SIZE, LAST_MODIFIED, TYPE
 }
 
-enum class FileActionType{
+enum class FileActionType {
     EDIT_AS_OPEN, EDIT_AND_OPEN, OPEN_ONLY, OPEN_ONLY_DETACH_PROCESS
 }
 
-private val acceptedImageExt = listOf(".jpg", ".jpeg", ".png")
-private val acceptedEditOnlyExt = arrayListOf(".txt", ".doc", ".docx", ".odt", "ods")
+const val PDF_PREVIEW_FOLDER_NAME = "pdf_preview"
+
+private val acceptedImageExt = listOf("jpg", "jpeg", "png")
+private val acceptedVideoExt = listOf("mp4", "avi", "mov", "wmv", "flv")
+private val acceptedEditOnlyExt = arrayListOf("txt", "doc", "docx", "odt", "ods")
     .also { it.addAll(acceptedImageExt) }
-private val acceptedReadAndEditExt = listOf(".pdf", ".md")
-private val acceptedVideoExt = listOf(".mp4", ".wmv", ".avi", ".flv", ".mkv")
+private val acceptedReadAndEditExt = listOf("pdf", "md")
 
-fun provideIconImage(file: Path): Path? =
-    providePreview(file) //todo downscale to, say, 128x128
+fun isImage(filePath: Path?): Boolean {
+    val extension = extension(filePath)
+    return acceptedImageExt.contains(extension)
+}
 
-//might be a temporary file
-fun providePreview(file: Path): Path? =
-    if (isImage(file)) {
-        file
+fun isVideo(filePath: Path?): Boolean {
+    return if (filePath?.toFile()?.exists() == true && filePath.fileSize() > 0) {
+        val extension = extension(filePath)
+        acceptedVideoExt.contains(extension)
     } else {
-        if (false) { //todo: create previews from videos, pdfs, etc.
-            TODO()
-        } else {
-            null
-        }
+        false
     }
+}
 
-fun isImage(filePath: Path): Boolean {
-    val extension = ".${extension(filePath)}"
-    return acceptedImageExt.contains(extension)
+fun isFormat(filePath: Path?, ext: String): Boolean {
+    return extension(filePath) == ext
 }
 
-fun getFileActionType(filePath: Path): FileActionType{
-    return when(".${extension(filePath)}"){
+fun getPdfPreviewsFolder(): File =
+    Paths.get("${App.instance.cacheDir}/$PDF_PREVIEW_FOLDER_NAME").toFile()
```

Comment:
Not necessary to convert to `java.io.File`, it's possible to use `java.nio.Path` almost everywhere.